pkg/util/vault.go

```diff
@@ -61,88 +62,83 @@ Example JSON structure in the KMS config is,
-	VaultAuthPath       string `json:"vaultAuthPath"`
-	VaultRole           string `json:"vaultRole"`
-	VaultNamespace      string `json:"vaultNamespace"`
-	VaultPassphraseRoot string `json:"vaultPassphraseRoot"`
-	VaultPassphrasePath string `json:"vaultPassphrasePath"`
-	VaultCAVerify       bool   `json:"vaultCAVerify"`
-	VaultCAFromSecret   string `json:"vaultCAFromSecret"`
+	EncryptionKMSID     string
+	VaultAddress        string
+	VaultAuthPath       string
+	VaultRole           string
+	VaultNamespace      string
+	VaultPassphraseRoot string
+	VaultPassphrasePath string
+	VaultCAVerify       bool
 	vaultCA             *x509.CertPool
 }
 
 // InitVaultKMS returns an interface to HashiCorp Vault KMS
-func InitVaultKMS(opts, secrets map[string]string) (EncryptionKMS, error) {
-	var config []VaultKMS
-
-	vaultID, ok := opts["encryptionKMSID"]
-	if !ok {
-		return nil, fmt.Errorf("missing encryptionKMSID for vault as encryption KMS")
-	}
-
-	// #nosec
-	content, err := ioutil.ReadFile(kmsConfigPath)
-	if err != nil {
-		return nil, fmt.Errorf("error fetching vault configuration for vault ID (%s): (%s)",
-			vaultID, err)
-	}
-
-	err = json.Unmarshal(content, &config)
-	if err != nil {
-		return nil, fmt.Errorf("unmarshal failed: %v. raw buffer response: %s",
-			err, string(content))
-	}
-
-	for i := range config {
-		vault := &config[i]
-		if vault.EncryptionKMSID != vaultID {
-			continue
-		}
-		if vault.VaultAddress == "" {
-			return nil, fmt.Errorf("missing vaultAddress for vault as encryption KMS")
-		}
-		if vault.VaultAuthPath == "" {
-			vault.VaultAuthPath = vaultDefaultAuthPath
-		}
-		if vault.VaultRole == "" {
-			vault.VaultRole = vaultDefaultRole
-		}
-		if vault.VaultNamespace == "" {
-			vault.VaultNamespace = vaultDefaultNamespace
-		}
-		if vault.VaultPassphraseRoot == "" {
-			vault.VaultPassphraseRoot = vaultDefaultPassphraseRoot
+func InitVaultKMS(kmsID string, config, secrets map[string]string) (EncryptionKMS, error) {
+	var (
+		ok  bool
+		err error
+	)
+	kms := &VaultKMS{}
+	kms.EncryptionKMSID = kmsID
+
+	kms.VaultAddress, ok = config["vaultAddress"]
+	if !ok || kms.VaultAddress == "" {
+		return nil, fmt.Errorf("missing 'vaultAddress' for vault KMS %s", kmsID)
+	}
+	kms.VaultAuthPath, ok = config["vaultAuthPath"]
+	if !ok || kms.VaultAuthPath == "" {
+		kms.VaultAuthPath = vaultDefaultAuthPath
+	}
+	kms.VaultRole, ok = config["vaultRole"]
+	if !ok || kms.VaultRole == "" {
+		kms.VaultRole = vaultDefaultRole
+	}
+	kms.VaultNamespace, ok = config["vaultNamespace"]
+	if !ok || kms.VaultNamespace == "" {
+		kms.VaultNamespace = vaultDefaultNamespace
+	}
+	kms.VaultPassphraseRoot, ok = config["vaultPassphraseRoot"]
+	if !ok || kms.VaultPassphraseRoot == "" {
+		kms.VaultPassphraseRoot = vaultDefaultPassphraseRoot
+	}
+	kms.VaultPassphrasePath, ok = config["vaultPassphrasePath"]
+	if !ok || kms.VaultPassphrasePath == "" {
+		kms.VaultPassphrasePath = vaultDefaultPassphrasePath
+	}
+	kms.VaultCAVerify = true
```

Comment:
do we need to enable CAVerify by default? if yes, can you add a comment here that we are enabling CAVerify by default

Reply:
in general all tools have SSL verification on by default, this is an option to be able to disable it if needed, for the code itself it remains as it was before, just a bit more explicit